### PR TITLE
Add prettier code formatting

### PR DESCRIPTION
**Pull request by sbrunner**
Examples on apply:
- https://github.com/camptocamp/ngeo/pull/6625
- https://github.com/camptocamp/c2cgeoportal/pull/7550
- https://github.com/mapfish/mapfish-print/pull/1656 (except java files)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,7 +2,8 @@ GITHUB_REPOSITORY ?= camptocamp/c2cciutils
 
 .PHONY: build
 build: checks
-		docker build --tag=$(GITHUB_REPOSITORY) .
+	cd c2cciutils; npm install
+	docker build --tag=$(GITHUB_REPOSITORY) .
```
Comment (arnaud-morvan) [inline Makefile:6]: Why not running npm install in Dockerfile ?
Seems you need npm packages for the packaging.
Maybe desserve a comment.

File: c2cciutils/checks.py [hunk 2]
```diff
@@ -1029,6 +1029,44 @@ def setup(
     return success
 
 
+def prettier(config, full_config, args):
+    """
+    Run prittier check on all the supported files
```
Comment (arnaud-morvan) [inline c2cciutils/checks.py:1034]:
```suggestion
    Run prettier check on all the supported files
```

File: c2cciutils/prettier.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        try:
+            return self.module.call_member("format", data, config)
+        except node_vm2.VMError as exception:
+            print(exception)
+            return data
+
+    def format(self, filename, config):
+        try:
+            with open(filename) as the_file_to_format:
+                new_data = self.module.call_member("format", the_file_to_format.read(), config)
```
Comment (arnaud-morvan) [inline c2cciutils/prettier.py:81]: What does this ?

File: test/test.css [hunk 4]
```diff
@@ -0,0 +1,5 @@
+.ol-box {
```
Comment (arnaud-morvan) [inline test/test.css:1]: I suppose those files are used for testing prettier, but I do not see any test using them.
=> Hard to understand.
Desserve a comment somewhere ?

Comment (sbrunner) [reply test/test.css:1]: Not used effectively, removed :-)

File: c2cciutils/prettier.py (hunk 3 above)
Comment (sbrunner) [reply c2cciutils/prettier.py:114]: The call_member on module will call the JavaScript functions...

File: c2cciutils/prettier.py [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+    def __init__(self, module: node_vm2.NodeVMModule) -> None:
+        """
+        Initialize.
+        """
+        self.module = module
+
+    def get_info(self, filename: str) -> Dict[str, Any]:
+        """
+        Get the Prettier infromations related to a file.
```
Comment (arnaud-morvan) [inline c2cciutils/prettier.py:29]:
```suggestion
        Get the Prettier informations related to a file.
```

File: c2cciutils/prettier.py [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+        Close.
+        """
+        assert self.module is not None
+
+        return self.module.module.__exit__(tpe, value, traceback)
+
+
+def dump_yaml(data: Dict[str, Any], filename: str = "ci/config.yaml") -> str:
+    """
+    Dump the YAML by respection the prettier configuration.
```
Comment (arnaud-morvan) [inline c2cciutils/prettier.py:167]:
```suggestion
    Dump the YAML according to the prettier configuration.
```

**Review by arnaud-morvan — APPROVED** [review]
Thanks, still some typos

**Pull request merged**